packages/contracts-bedrock/test/vendor/eas/SchemaRegistry.t.sol

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import { Test } from "forge-std/Test.sol";
+import { console } from "forge-std/console.sol";
+import { ISchemaRegistry, SchemaRecord } from "src/vendor/eas/ISchemaRegistry.sol";
+import { ISchemaResolver } from "src/vendor/eas/resolver/ISchemaResolver.sol";
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { CommonTest } from "test/setup/CommonTest.sol";
+
+contract SchemaRegistryTest is CommonTest {
+    // State variables
+    ISchemaRegistry registry;
+
+    // Events
+    event Registered(
+        bytes32 indexed uid,
+        address indexed registerer,
+        SchemaRecord schema
+    );
+
+    function setUp() public override {
+        super.setUp();  // Call parent setUp first
+        registry = ISchemaRegistry(Predeploys.SCHEMA_REGISTRY);  // Get registry from predeploy
+    }
+
+    // Helper functions
+    function getUID(
+        string memory schema,
+        address resolver,
+        bool revocable
+    ) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(schema, resolver, revocable));
+    }
+
+    // Should be equal to the current version of SchemaRegistry.sol in this repository
+    function testVersion() public view {
+     assertEq(ISemver(address(registry)).version(), "1.3.1-beta.1");
+    }
+
+    // Basic functionality tests
+    function testRegisterSchema(string memory _schema, address _resolver, bool _revocable) public {
+
+        bytes32 uid = registry.register(_schema, ISchemaResolver(_resolver), _revocable);
+        SchemaRecord memory record = registry.getSchema(uid);
+        
+        assertEq(record.uid, uid);
+        assertEq(record.schema, _schema);
+        assertEq(address(record.resolver), _resolver);
+        assertEq(record.revocable, _revocable);
+    }
+
+    function testGetSchema() public {
+        string memory schema = "bool isFriend";
+        address resolver = address(0x456);
+        bool revocable = true;
+
+        bytes32 uid = registry.register(schema, ISchemaResolver(resolver), revocable);
+        SchemaRecord memory record = registry.getSchema(uid);
+
+        assertEq(record.uid, uid);
+        assertEq(record.schema, schema);
+        assertEq(address(record.resolver), resolver);
+        assertEq(record.revocable, revocable);
+    }
+
+    // Edge cases
+    function testRegisterSchemaWithoutSchema() public {
+        string memory schema = "";
+        address resolver = address(0x123);
```

Comment:
This should be a fuzz test for resolver and revocable but hardcode `""` for schema